### PR TITLE
Trace memory alloc/dealloc by thread

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What is changed and how it works?

Issue Number: Close #xxx

What's Changed:

```commit-message

```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note

```

### CODE REVIEW

File: proxy_components/proxy_server/src/run.rs [hunk 1]
```diff
@@ -618,10 +618,27 @@ impl<ER: RaftEngine, F: KvFormat> TiKvServer<ER, F> {
...
+        // NOTE: Compat disagg arch upgraded from * to 8.0.
+        {
+            let raft_engine_path = config.raft_engine.config().dir + "/ps_engine";
+            let path = Path::new(&raft_engine_path);
+            if path.exists() {
+                info!("created ps_engine.raftlog for upgraded cluster");
+                let new_raft_engine_path = config.raft_engine.config().dir + "/ps_engine.raftlog";
+                std::fs::File::create(new_raft_engine_path).unwrap();
+            }
+        }
```
Comment (JaySon-Huang) [inline proxy_components/proxy_server/src/run.rs:640]: Fix https://github.com/pingcap/tiflash/issues/8589

File: proxy_components/proxy_server/src/run.rs [hunk 2]
```diff
@@ -618,10 +618,27 @@ impl<ER: RaftEngine, F: KvFormat> TiKvServer<ER, F> {
...
+
+        // NOTE: Compat disagg arch upgraded from * to 8.0.
+        {
+            let raft_engine_path = config.raft_engine.config().dir + "/ps_engine";
+            let path = Path::new(&raft_engine_path);
+            if path.exists() {
+                info!("created ps_engine.raftlog for upgraded cluster");
+                let new_raft_engine_path = config.raft_engine.config().dir + "/ps_engine.raftlog";
+                std::fs::File::create(new_raft_engine_path).unwrap();
+            }
```
Comment (JaySon-Huang) [inline proxy_components/proxy_server/src/run.rs:639]: Only create the file and print the logging when `path.exists() && !new_raft_engine_path.exists()`?

File: proxy_components/proxy_server/src/run.rs (hunk 1 above)
Comment (CalvinNeo) [reply proxy_components/proxy_server/src/run.rs:640]: Will use a sep PR

File: proxy_components/proxy_server/src/run.rs (hunk 2 above)
Comment (CalvinNeo) [reply proxy_components/proxy_server/src/run.rs:639]: If we found ps_engine, we will create a ps_engine.raftlog

**Pull request merged**